nessie/src/main/java/org/apache/iceberg/nessie/ParsedTableIdentifier.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.nessie;
+
+import com.dremio.nessie.client.NessieClient;
+import java.time.Instant;
+import java.util.Map;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+public class ParsedTableIdentifier {
```

Comment:
I'm wondering if there is a more specific name for this class. Maybe something like `TableReference` because it has both an identifier and a ref? Or maybe `NessieIdentifier`? `ParsedTableIdentifier` doesn't really tell me what is different about this as opposed to `TableIdentifier`.

Reply:
I chose `TableReference` in the end. I don't love it but its way better than `ParsedTableIdentifier`. I don't know how to concisely state that its a `TableIdentifier` tied to a specific `Reference`. I have updated the javadoc which I hope clarifies it